src/main/kotlin/io/ionic/libs/ionfilesystemlib/helper/common/IONFILEUriResolveExtensions.kt

```diff
@@ -0,0 +1,67 @@
+package io.ionic.libs.ionfilesystemlib.helper.common
+
+import io.ionic.libs.ionfilesystemlib.helper.IONFILEUriHelper
+import io.ionic.libs.ionfilesystemlib.model.IONFILEExceptions
+import io.ionic.libs.ionfilesystemlib.model.IONFILEUri
+import io.ionic.libs.ionfilesystemlib.model.LocalUriType
+
+internal suspend inline fun <reified T> IONFILEUriHelper.useUriIfResolvedAsLocalDirectory(
+    uri: IONFILEUri,
+    onResolved: (IONFILEUri.Resolved.Local) -> Result<T>
+): Result<T> = useUriIfResolvedAsLocal(uri) { resolvedUri ->
+    if (resolvedUri.type != LocalUriType.FILE) {
+        onResolved(resolvedUri)
+    } else {
+        Result.failure(IONFILEExceptions.NotSupportedForFiles())
+    }
+}
+
+internal suspend inline fun <reified T> IONFILEUriHelper.useUriIfResolvedAsLocalFile(
+    uri: IONFILEUri,
+    onResolved: (IONFILEUri.Resolved.Local) -> Result<T>
+): Result<T> = useUriIfResolvedAsLocal(uri) { resolvedUri ->
+    if (resolvedUri.type != LocalUriType.DIRECTORY) {
```

Comment:
Same question as above, can't we check for `FILE` instead? Or do we accept it it is `UNKNOWN` ?